src/Optimization/hiopAlgFilterIPM.cpp

```diff
@@ -1485,6 +1529,170 @@ void hiopAlgFilterIPMQuasiNewton::outputIteration(int lsStatus, int lsNum, int u
+    nlp->log->printf(hovError,  "  Addtl info: %s\n", exp.what());
+    success = false;    
+  }  
+  delete ds;
+  return success;
+}
+
+bool hiopAlgFilterIPMQuasiNewton::load_state_from_file(const ::std::string& path) noexcept
+{
+  sidre::DataStore* ds = nullptr;
+  auto success = true;
+  try {
+    ds = new sidre::DataStore();
+    sidre::IOManager reader(this->get_nlp()->get_comm());
+    reader.read(ds->getRoot(), path, false);
+    
+    this->load_state_from_data_store(ds);
+    
+  } catch(const std::exception& exp) {
+    nlp->log->printf(hovError, "Error in loading checkpoint from file '%s'\n", path.c_str());
+    nlp->log->printf(hovError,  "  Addtl info: %s\n", exp.what());
+    success = false;
+  }
+  delete ds;
+  return success;
+}
+
+void hiopAlgFilterIPMQuasiNewton::
+copy_vec_to_new_view(const ::std::string& name, const hiopVector* vec, sidre::Group* nlp_group)
+{
+  using IndType = sidre::IndexType;
+  const IndType size = vec->get_local_size();
+  sidre::View* dest = nlp_group->createViewAndAllocate(name, sidre::DOUBLE_ID, size);
+  
+  const auto stride(dest->getStride());
+  double *const dest_ptr(dest->getArray());
+  const double* arr = vec->local_data_host_const();
+  if(1==stride) {
+    ::std::copy(arr, arr+size, dest_ptr);
+  } else {
+    for(IndType i=0; i<size; ++i) {
+      dest_ptr[i*stride] = arr[i];
+    }
+  }  
+}
+void hiopAlgFilterIPMQuasiNewton::
+copy_vec_from_view(const ::std::string& name, hiopVector* vec, const sidre::Group* nlp_group)
+{
+  const sidre::View* view_const = nlp_group->getView(name);
+  if(!view_const) {
+    ::std::stringstream ss;
+    ss << "Could not find view '" << name << "in sidre::Group.\n";
+    throw ::std::runtime_error(ss.str());
+  }
+  // const_cast becase View does not have a const getArray()
+  sidre::View* view = const_cast<sidre::View*>(view_const);  
+  if(view) {
+    const hiop::size_type size = vec->get_local_size();
+    if(view->getNumElements() != size) {
+      ::std::stringstream ss;
+      ss << "Size mismatch for state/view '" << name << "' between hiop and sidre::View: " <<
+        " HiOp state is " << size << " doubles, while the view has " << view->getNumElements() <<
+        " double elements.\n";
+      throw ::std::runtime_error(ss.str());
+    }
+    double* arr_dest = vec->local_data_host();
+    const double *const arr_src = view->getArray();
+    const auto stride(view->getStride());
+    if(1==stride) {
+      ::std::copy(arr_src, arr_src+size, arr_dest);
+    } else {
+      for(hiop::index_type i=0; i<size; ++i) {
+        arr_dest[i] = arr_src[i*stride];
+      }
+    }    
+  } else {
+    assert(false && "const cast failed for Sidre::View");
+  }
+}
+
+void hiopAlgFilterIPMQuasiNewton::save_state_to_data_store(::axom::sidre::DataStore* ds)
+{
+  using IndType = sidre::IndexType;
+  sidre::Group* nlp_group = ds->getRoot()->createGroup("hiop solver");
+  
+  //create views for each member that needs to be saved
```

Comment:
It might be useful to include a little metadata too such as:
- the version of HiOp that's writing the file
- the iteration number?